joblib/_store_backends.py

```diff
@@ -384,29 +384,29 @@ def get_items(self):
 
         return items
 
-    def configure(self, location, verbose=1, backend_options={}):
+    def configure(self, location, verbose=1, backend_options=None):
         """Configure the store backend.
 
         For this backend, valid store options are 'compress' and 'mmap_mode'
         """
+        if backend_options is None:
+            backend_options = {}
 
         # setup location directory
         self.location = location
         if not os.path.exists(self.location):
             mkdirp(self.location)
 
         # item can be stored compressed for faster I/O
-        self.compress = backend_options['compress']
+        self.compress = backend_options.get('compress', False)
```

Comment:
Since `mmap_mode` is optional in `backend_options` I think it makes sense to have the same status for `compress`.